block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -322,4 +330,47 @@ object BlockDagKeyValueStorage {
       stores.invalidBlocks,
       stores.equivocations
     )
+
+  def migrateLfb[F[_]: Sync: Log](
+      lastFinalizedStorage: LastFinalizedStorage[F],
+      kvm: KeyValueStoreManager[F]
+  ): F[Unit] = {
+    val errNoLfbInStorage   = "No LFB in LastFinalizedStorage when attempting migration."
+    val errNoMetadataForLfb = "No metadata found for LFB when attempting migration."
+    val errNoMetadataForFb  = "No metadata found for finalized block when attempting migration."
+
+    for {
+      blockMetadataDb <- kvm.database[BlockHash, BlockMetadata](
+                          "block-metadata",
+                          codecBlockHash,
+                          codecBlockMetadata
+                        )
+      // record LFB
+      lfb  <- lastFinalizedStorage.get() >>= (_.liftTo(new Exception(errNoLfbInStorage)))
+      curV <- blockMetadataDb.get(lfb).flatMap(_.liftTo[F](new Exception(errNoMetadataForLfb)))
+      _    <- blockMetadataDb.put(lfb, curV.copy(directlyFinalized = true, finalized = true))
+      blocksInfoMap <- blockMetadataDb
+                        .collect {
+                          case (hash, metaData) =>
+                            (hash, BlockMetadataStore.blockMetadataToInfo(metaData()))
+                        }
+                        .map(_.toMap)
+      _ <- Log[F].info("Migration of LFB done.")
+
+      // record finalized blocks
+      finalizedBlockSet = DagOps
+        .bfTraverse(List(lfb)) { bh =>
+          blocksInfoMap.get(bh).map(_.parents.toList).getOrElse(List.empty)
+        }
+        .toList
+      curVs <- finalizedBlockSet.traverse { h =>
+                blockMetadataDb.get(h).flatMap(_.liftTo[F](new Exception(errNoMetadataForFb)))
```

Comment:
Not sure about how the performance of migrating the database. If that takes too long. We'd better parallel reading here to be faster.

Reply:
I'd add to read in chunks because this will load all DAG store with around ~5GB data. Also the whole chunk can be loaded with list overload of _get_.
Sorted keys can also speed up write but I'm not completely sure if ordering is the same.

Reply:
That's better

Reply:
Do you have an idea what chunk size would be preferable?

Reply:
10k looks reasonable, ptal at description.